src/zarr/storage/_obstore.py

```diff
@@ -224,7 +224,7 @@ async def set_if_not_exists(self, key: str, value: Buffer) -> None:
         self._check_writable()
         buf = value.to_bytes()
         with contextlib.suppress(obs.exceptions.AlreadyExistsError):
-            await obs.put_async(self.store, key, buf, mode="create")
+            await obs.put_async(self.store, key, buf, mode="Create")
```

Comment:
I don't think this should change anything. We need a better reproduction of the original error I think